request.go

```diff
@@ -79,6 +133,32 @@ func (c *Client) ListRequests() (*Requests, error) {
 	return ret, err
 }
 
+// ListRequestsWithFilter lists all requests that matches the given filters
+// Available filters are:
+// * url
+// * createdDate
+// * method
+// * body
+// * requestStatus
+func (c *Client) ListRequestsWithFilter(filter *RequestListFilter) (*Requests, error) {
+	path := "/requests"
+	query := url.Values{
+		"depth": []string{"10"},
```

Comment:
From https://devops.ionos.com/api/cloud/v4/#requests:

> You can increase the amount of detail returned in some requests by supplying an integer value between 1 and 10 as depth:
https://api.ionos.com/cloudapi/v4/requests?depth=1
Note: For this particular operation, increasing the value supplied to depth will not return additional information about the requests.

Sounds like 10 is not wrong, but 1 is enough.

Reply:
Actually I would simply use depth=10 everywhere. I don't see any point in using different depth settings for different api calls which would then return more or less details.